src/check_result.jl

```diff
@@ -56,19 +80,32 @@ function check_equal(
     @test ActualPrimal === ExpectedPrimal
 end
 
-# This catches comparisons of Composites and Tuples/NamedTuple
-# and gives a error messaage complaining about that
-check_equal(::C, expected::T) where {C<:Composite, T} = @test C === T
-check_equal(::T, expected::C) where {C<:Composite, T} = @test C === T
 
+# Some structual differential and a natural differential
+function check_equal(actual::Composite{P, T}, expected; kwargs...) where {T, P}
+    if _can_pass_early(actual, expected)
+        @test true
+    else
+        @assert (T <: NamedTuple)  # it should be a structual differential if we hit this
 
-check_equal(::Zero, x; kwargs...) = check_equal(zero(x), x; kwargs...)
-check_equal(x, ::Zero; kwargs...) = check_equal(x, zero(x); kwargs...)
-check_equal(x::Zero, y::Zero; kwargs...) = @test true
+        # We are only checking the properties that are in the Composite
+        # the natural differential is allowed to have other properties that we ignore
```

Comment:
What other properties could it have? Maybe becomes clearer once I know what a natural differential is

Reply:
All kinds of junk potentially.
Like metadata; or reduncant precomputed values.
But i am also particularly thinking of Factorizations.